core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -40,9 +40,11 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
         status
       }
     }
+
+    val numApps = Option(limit).getOrElse(Integer.MAX_VALUE).asInstanceOf[Int]
```

Comment:
Oh I see, this is weird because `limit` is a `java.lang.Integer`. I suggest `Option(limit).map(_.toInt).getOrElse(Int.MaxValue)` is a tiny bit better.